### PR TITLE
[basic] Replace \term with \defnx as appropriate

### PR DESCRIPTION
**Pull request by jensmaurer**
Partially addresses #329.

Also removes \term from the introductory note in 3p1 and uses \defnx instead of separate \term and \indextext where applicable.

diffpdf shows only whitespace differences in the main body, plus one slightly moved footnote. The index has a few more entries now, e.g. for "object, const".

### CODE REVIEW

**Comment by jensmaurer** [comment]
One \term{object type} remains, because there is a separate issue about incompatible duplicate definitions of "object type".

**Comment by tkoeppe** [comment]
Are you sure this doesn't clash with #980?

File: source/basic.tex [hunk 1]
```diff
@@ -4207,13 +4182,12 @@
...
-In this International Standard, the notation \term{cv} (or
-\term{cv1}, \term{cv2}, etc.), used in the description of types,
+In this International Standard, the notation \cv{} (or
+\cvqual{cv1}, \cvqual{cv2}, etc.), used in the description of types,
 represents an arbitrary set of cv-qualifiers, i.e., one of
 \{\tcode{const}\}, \{\tcode{volatile}\}, \{\tcode{const},
 \tcode{volatile}\}, or the empty set.
-\indextext{cv-qualifier!top-level}
-For a type \cv\ \tcode{T}, the \term{top-level cv-qualifiers}
+For a type \cv\ \tcode{T}, the \defnx{top-level cv-qualifiers}{cv-qualifier!top-level}
```
Comment (tkoeppe) [inline source/basic.tex]: Ditto, `\cv{}`.

**Comment by tkoeppe** [comment]
Very nice change, I'll be happy to merge this pending comment resolution.

**Comment by jensmaurer** [comment]
Sorry for the duplicate work; this is likely to clash / be redundant with #980.

File: source/basic.tex [hunk 2]
```diff
@@ -3595,8 +3589,8 @@
 The \defn{object representation}
 \indextext{representation!object}%
 of an object of type \tcode{T} is the
-sequence of \term{N} \tcode{unsigned} \tcode{char} objects taken up
-by the object of type \tcode{T}, where \term{N} equals
+sequence of \placeholder{N} \tcode{unsigned} \tcode{char} objects taken up
```
Comment (jensmaurer) [inline source/basic.tex]: I can't see the "N" here being a code placeholder.

File: source/basic.tex [hunk 3]
```diff
@@ -3609,16 +3603,16 @@
...
 contexts~(\ref{dcl.enum}), or an array of unknown
-size or of incomplete element type, is an \defn{incompletely-defined object
-type}.\footnote{The size and layout of an instance of an incompletely-defined
+size or of incomplete element type, is an
+\defnx{incompletely-defined object type}{object type!incompletely-defined}.%
+\footnote{The size and layout of an instance of an incompletely-defined
 object type is unknown.}
-Incompletely-defined object types and \cv\ \tcode{void} are \term{incomplete
-types}~(\ref{basic.fundamental}). Objects shall not be defined to have an
+Incompletely-defined object types and \cv\ \tcode{void} are
```
Comment (jensmaurer) [inline source/basic.tex]: I don't know. The "\cv\ " was already there, and I didn't touch it (because I was replacing \term).
If we want to do the replacement, it's a script job (29 occurrences).

**Comment by tkoeppe** [comment]
That's OK. I like this PR because it is small and understandable. I'd be happy to apply this; the others need rebasing anyway, so identical, duplicate changes will just go away.

File: source/basic.tex (hunk 3 above)
Comment (tkoeppe) [inline source/basic.tex]: It's not so much that we actively want to spend time on it, but if there's a chance for a drive-by fix, it's usually good to make small improvements. On the other hand, if you have limitless time and would like to do things systematically, even better :-)
